amplify/backend/function/getUserTokenBalance/src/utils.js

```diff
@@ -47,15 +44,27 @@ const getStakedTokens = async (
 ) => {
   const { data } =
     (await graphqlRequest(
-      getColonyStake,
+      getUserStakesInColony,
       {
-        colonyStakeId: getColonyStakeId(walletAddress, colonyAddress),
+        userAddress: walletAddress,
+        colonyAddress,
       },
       apiKey,
       graphqlURL,
     )) ?? {};
 
-  return data?.getColonyStake?.totalAmount ?? 0;
+  let totalAmount = BigNumber.from(0);
+  if (data?.getUserStakesInColony) {
+    data.getUserStakesInColony.items.forEach((item) => {
+      totalAmount = totalAmount.add(item.amount);
+    });
+  } else {
+    console.error(
+      `Could not get user ${walletAddress} stakes in colony ${colonyAddress}`,
+    );
+  }
+
+  return totalAmount.toString();
```

Comment:
Given we use a limit in the getUserStakesInColony results, shouldn't we make use of the pagination feature to ensure the amount is properly computed?

Reply:
That's a fair point, I thought about it but then came to a conclusion it's pretty unlikely for any user to have more than 10,000 stakes, which is when pagination would be required. 

It's great that you're keeping an eye out on this, if there wasn't @index on userAddress, amplify would get 10,000 records (for any user in any colony), and only then apply filtering, and we would hit pagination issues much sooner.

Reply:
Fair, was thinking the limit is high enough ftm, just to keep in mind to implement pagination at some point to be sure we will not run into issues in the future

Reply:
@mmioana would you mind approving? Thanks!